tools/journal_proto_processor.cpp

```diff
@@ -1891,18 +1920,46 @@ void JournalProtoProcessor::ProcessMethodExtension(
   }
   cxx_functions_implementation_[descriptor] += cxx_run_epilog + "}\n\n";
 
-  cs_interface_method_declaration_[descriptor] = Join(
-      {"  [DllImport(dllName           : dll_path,\n"
-       "             EntryPoint        = \"principia__" + name + "\",\n"
-       "             CallingConvention = CallingConvention.Cdecl)]",
-       cs_interface_return_marshal,
-       "internal static extern " + cs_interface_return_type + " " + name + "("},
-      "\n  ");
+  cs_interface_method_declaration_[descriptor] =
```

Comment:
This needs a new identifier because, believe it or not, it's not a method anymore (hint: the word `class`).  I object to just lumping text in this variable because it's convenient.  Let's keep the data structures moderately clean.  (And yes, some new code will be needed to output this new map.)

I would by the way be fine with <all the partial classes> <all the interface methods> in the generated code.  I don't think there is value to glueing the two together.

Reply:
Well, if they are lumped together, they do not need to be separate partial classes.

Reply:
Split the maps but kept the interleaving.